inspire_dojson/hep/rules/bdFFT.py

```diff
@@ -45,32 +46,49 @@ def _is_hidden(value):
         return 'HIDDEN' in [val.upper() for val in value] or None
 
     def _get_caption(value):
-        return ''.join(re.compile('^\d{5}\s+').split(value)[1:]) or None
+        return ''.join(re.compile('^\d{5}\s*').split(value)[1:]) or None
+
+    def _get_key(value):
+        fname = value.get('n', str(uuid.uuid4()))
+        extension = value.get('f', '')
+
+        if fname.endswith(extension):
+            return fname
+        return fname + extension
+
+    ordered_pattern = re.compile('^(\d{5})\s*')
```

Comment:
why do you need two patterns? it would be simpler to modify `_get_caption` to `_get_index_and_caption` that would return both the `index` (e.g. `00005` or `None`) and the `caption`.